src/jitlayers.cpp

```diff
@@ -336,9 +339,21 @@ void JuliaOJIT::DebugObjectRegistrar::operator()(RTDyldObjHandleT H,
                    static_cast<const RuntimeDyld::LoadedObjectInfo*>(&LOS));
 }
 
+struct CompileTiming {
+  CompileTiming(std::string name) :name(name) {
+      JULIA_COMPILE_START();
+  }
+  ~CompileTiming() {
+      JULIA_COMPILE_END(name.c_str());
+  }
+  std::string name;
+};
 
 object::OwningBinary<object::ObjectFile> JuliaOJIT::CompilerT::operator()(Module &M)
 {
+    // Instrument the time spent in this block for this method instance
+    CompileTiming _c(M.getName().str());
```

Comment:
@JeffBezanson  do you know where I can find the caller of this function in a context that would have the name of the method instance?

Reply:
Ah, okay, i put a breakpoint and got a stacktrace, and it gives me:

```
    frame #4: 0x00000001001b6ab7 libjulia.1.2.dylib`JuliaOJIT::addModule(this=<unavailable>, M=unique_ptr<llvm::Module, std::__1::default_delete<llvm::Module> > @ 0x00007ffeefbfe630) at jitlayers.cpp:477:41 [opt]
    frame #5: 0x00000001001b73a3 libjulia.1.2.dylib`jl_finalize_function(llvm::StringRef) [inlined] jl_add_to_ee(m=unique_ptr<llvm::Module, std::__1::default_delete<llvm::Module> > @ scalar) at jitlayers.cpp:702:25 [opt]
    frame #6: 0x00000001001b738f libjulia.1.2.dylib`jl_finalize_function(F=<unavailable>) at jitlayers.cpp:710 [opt]
    frame #7: 0x00000001001596e9 libjulia.1.2.dylib`getAddressForFunction(fname=<unavailable>) at codegen.cpp:1338:5 [opt]
    frame #8: 0x0000000100159a77 libjulia.1.2.dylib`::jl_generate_fptr(output=0x000000010c443c10) at codegen.cpp:1432:39 [opt]
    frame #9: 0x00000001000ddb01 libjulia.1.2.dylib`jl_compile_method_internal(mi=0x000000010c524350, world=25987) at gf.c:1795:5 [opt]
    frame #10: 0x00000001000e02ab libjulia.1.2.dylib`jl_apply_generic(args=0x00007ffeefbfe790, nargs=2) at gf.c:2196:16 [opt]
```

But i'm surprised; isn't `finalize` happening _after_ the optimization passes have finished? Why is the `JL_TIMING(LLVM_OPT);` block there?